external-crates/move/crates/move-compiler/src/typing/translate.rs

```diff
@@ -2937,12 +2938,88 @@ fn module_call_impl(
     (call, return_)
 }
 
+/// If the constant that we are referencing has an `error` attribute, we need to change the type of
+/// the constant to a u64 since this will be compiled into a u64 error code.
+fn annotated_error_const(context: &mut Context, e: &mut T::Exp) {
+    let sp!(
+        const_loc,
+        T::UnannotatedExp_::Constant(module_ident, constant_name)
+    ) = &mut e.exp
+    else {
+        return;
+    };
+
+    let ConstantInfo {
+        attributes,
+        defined_loc,
+        signature,
+    } = context.constant_info(module_ident, constant_name);
+    let has_error_annotation = attributes.contains_key_(&known_attributes::ErrorAttribute.into());
+    let is_u64_type = matches!(
+        signature.value.builtin_name(),
+        Some(sp!(_, N::BuiltinTypeName_::U64))
+    );
+    let ty_loc = e.ty.loc;
+    let const_def_loc = *defined_loc;
+
+    if has_error_annotation {
+        context.env.check_feature(
+            context.current_package(),
+            FeatureGate::CleverAssertions,
+            *const_loc,
+        );
+    }
```

Comment:
Thinking about it more, I think we should just add the feature gate while resolving the attribute?

Then we can remove any and all gating here